crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs

```diff
@@ -458,6 +460,50 @@ impl OutboundGroupSession {
         session.encrypt(&plaintext)
     }
 
+    /// Encrypt an arbitrary event for the given room.
+    ///
+    /// Beware that a room key needs to be shared before this method
+    /// can be called using the `share_room_key()` method.
+    ///
+    /// # Arguments
+    ///
+    /// * `payload` - The plaintext content of the event that should be
+    ///   encrypted in raw JSON form.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the content can't be serialized.
+    async fn encrypt_inner<T: Serialize>(
+        &self,
+        payload: &T,
```

Comment:
Nice idea to take a generic `T: Serialize` here, though I'm slightly hesitant because IIRC this means we'll end up with a separate copy of `encrypt_inner` for each `Payload` type. It's probably fine though.